sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_deployment_base_model.py

```diff
@@ -0,0 +1,123 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+# pylint: disable=protected-access,arguments-renamed
+
+import logging
+from abc import abstractmethod
+from os import PathLike
+from typing import IO, Any, AnyStr, Dict, Optional, Union
+
+from azure.ai.ml._restclient.v2022_02_01_preview.models import BatchDeploymentData
+from azure.ai.ml._restclient.v2022_05_01.models import OnlineDeploymentData
+from azure.ai.ml._utils.utils import dump_yaml_to_file
+from azure.ai.ml.entities._mixins import RestTranslatableMixin
+from azure.ai.ml.entities._resource import Resource
+from azure.ai.ml.exceptions import (
+    DeploymentException,
+    ErrorCategory,
+    ErrorTarget,
+)
+
+module_logger = logging.getLogger(__name__)
+
+
+class BatchDeploymentBaseModel(Resource, RestTranslatableMixin):
```

Comment:
This will also need to be the baseclass for `BatchDeployment` (otherwise the `type` field won't work, as it wont always be available).
It will also need to be backwards compatible with that API - we can do this using a `__getattr__` method rather than explicit properties - here's an example of how to do that:
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/azure/core/rest/_helpers.py#L199-L216

In terms of what we might return for these properties, fortunately I think most of these were `Optional`, so we could probably return `None`.